### PR TITLE
Updating ACLs to divide assignExercise permission between group and exercise, which can be checked independently.

### PR DESCRIPTION
**Pull request by krulis-martin**

### CODE REVIEW

File: app/config/permissions.neon [hunk 1]
```diff
@@ -356,24 +356,6 @@ permissions:
...
-      conditions:
-          - exercise.isSubGroupSupervisor
-          - group.isNotArchived
-
-    - allow: true
-      resource: group
-      actions:
-          - assignExercise
-      conditions:
-          - exercise.isPublic
```
Comment (Neloop) [inline app/config/permissions.neon:376]: Seems like this condition was lost... is that correct?

File: app/model/repository/Groups.php [hunk 2]
```diff
@@ -144,6 +150,54 @@ private function fetchGroupsByMembership(User $user, bool $admins): array
...
+        if (!$parentsIds) {
+            return [];
+        }
+        $qb = $this->createQueryBuilder('g'); // takes care of softdelete cases
+        $qb->andWhere($qb->expr()->in("g.parentGroup", $parentsIds));
+        return $qb->getQuery()->getResult();
+    }
+
+    /**
+     * Find all groups in are supervised by a user (as supervisor, admin, or admin inherently).
```
Comment (Neloop) [inline app/model/repository/Groups.php:170]: A bit weird formulation, `in` does not seem to fit there...

File: app/config/permissions.neon (hunk 1 above)
Comment (krulis-martin) [reply app/config/permissions.neon:376]: It is. New `exercise.assign` permission embeds this condition in `isSubGroupNonStudentMember` and `isSuperGroupAdmin`, although it is not visible in the permissions file.

**Review by Neloop — DISMISSED** [review]

File: app/model/repository/Groups.php (hunk 2 above)
Comment (krulis-martin) [reply app/model/repository/Groups.php]: Fixed.

**Review by Neloop — APPROVED** [review]

**Pull request merged**